### PR TITLE
Remove Spack MKL hack for older version of QMCPACK.

### PR DESCRIPTION
**Pull request by naromero77**
A Spack hack for MKL usage was needed in QMCPACK prior to version 3.5 when non-Intel compilers were used. This Spack hack could have undesirable side-effects and led to ugly code that was extra work to maintain. For older version of QMCPACK, we through a conflict if you want to use Intel MKL with the other compilers commonly found on x86.

Note that there is no impact to version of QMCPACK since 3.5.

### CODE REVIEW

**Comment by naromero77** [comment]
Could someone please review?

**Comment by naromero77** [comment]
@davydden @adamjstewart Any of you able to review and then merge?

File: var/spack/repos/builtin/packages/qmcpack/package.py [hunk 1]
```diff
@@ -92,6 +92,19 @@ class Qmcpack(CMakePackage, CudaPackage):
     conflicts('%pgi@:17', when='@3.6.0:', msg=compiler_warning)
     conflicts('%llvm@:3.4', when='@3.6.0:', msg=compiler_warning)
 
+    # Prior to QMCPACK 3.5.0 Intel MKL was not properly detected with
+    # non-Intel compilers without a Spack-based hack. This hack
+    # had the potential to for negative side effects and let to more
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/qmcpack/package.py]: typo: to for -> for

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/qmcpack/package.py]: typo: let -> led

File: var/spack/repos/builtin/packages/qmcpack/package.py [hunk 2]
```diff
@@ -92,6 +92,19 @@ class Qmcpack(CMakePackage, CudaPackage):
...
+    # complex Python code that would have been difficult to maintain
+    # long term. Note that this has not been an issue since QMCPACK 3.5.0.
+    # For older versions of QMCPACK, we issue a conflict below if you
+    # try to use Intel MKL with a non-Intel compilers.
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/qmcpack/package.py]: typo: compilers -> compiler

File: var/spack/repos/builtin/packages/qmcpack/package.py [hunk 3]
```diff
@@ -92,6 +92,19 @@ class Qmcpack(CMakePackage, CudaPackage):
...
+    mkl_warning = 'QMCPACK release prior to 3.5.0 require the ' \
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/qmcpack/package.py]: typo: release -> releases

File: var/spack/repos/builtin/packages/qmcpack/package.py [hunk 4]
```diff
@@ -258,35 +271,27 @@ def cmake_args(self):
...
             '-DLAPACK_LIBRARIES=%s' % lapack_blas.joined(';')
         ])
 
-        # Additionally, we need to pass the BLAS+LAPACK include directory for
-        # header files. This is to insure vectorized math and FFT libraries
-        # get properly detected. Intel MKL requires special case due to
-        # differences in Darwin vs. Linux $MKLROOT naming schemes. This section
-        # of code is intentionally redundant for backwards compatibility.
+        # Next two environment variables were introduced in QMCPACK 3.5.0
+        # Prior to v3.5.0, these lines should be benign but you CMake
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/qmcpack/package.py]: typo: you -> your (or just delete)

**Comment by naromero77** [comment]
@adamjstewart Thanks for the vote confidence :)

I am trying to see the one failure on codecov/project/build_systems but when I click the details link, I am just getting the Mac pin wheel and a timeout. Is there a real failure?

**Comment by adamjstewart** [comment]
@naromero77 see the note at the bottom of https://spack.readthedocs.io/en/latest/contribution_guide.html#coverage

**Comment by naromero77** [comment]
@adamjstewart thanks for the clarification.

**Pull request merged**